src/sparsezoo/analysis/helpers.py

```diff
@@ -0,0 +1,258 @@
+]
+
+
+def get_initializer(model: ModelProto, initializer_name: str) -> np.ndarray:
+    """
+    Helper function to find initializers by name in model graph
+
+    :param model: model used to find initializer
+    :param initializer_name: name of initializer being returned
+    :return: initalizer if found, None otherwise
+    """
+    for initializer in model.graph.initializer:
+        if initializer.name == initializer_name:
+            return initializer
+
+    return None
+
+
+def get_node_sparsity_sizes(model: ModelProto, node: NodeProto) -> Tuple[int, int]:
+    """
+    :param model: model in which node's parameter exists
+    :param node: node whose number of zeros and parameter size is being calculated
+    :return: number of zeros
+    :return: number of total values in node's parameter
+    """
+    zero_point = get_zero_point(model, node)
+    param = get_layer_param(model, node)
+    if param is None:
+        return 0, 0
+
+    num_zeros = np.count_nonzero(param == zero_point)
+
+    return num_zeros, param.size
+
+
+def get_node_four_block_sparsity_sizes(
+    model: ModelProto, node: NodeProto
+) -> Tuple[int, int]:
+    """
+    :param model: model in which node's parameter exists
+    :param node: node whose four block sparsity sizes are being calculated
+    :return: number of zero blocks
+    :return: number of total blocks
+    """
+    # Get param and zero point
+    zero_point = get_zero_point(model, node)
+    param = get_layer_param(model, node)
+    if param is None:
+        return 0, 0
+
+    # Bool array
+    param_zeros = param == zero_point
+
+    # Transpose so input channels are first
+    transpose_arg = np.arange(param_zeros.ndim)
+    transpose_arg[0] = 1
+    transpose_arg[1] = 0
+    param_zeros = np.transpose(param_zeros, transpose_arg)
+
+    # Flatten and treat weights as features across input channels
+    param_zeros = np.reshape(param_zeros, (param_zeros.shape[0], -1))
+
+    # Pad weight features with zeros to be divisible by four
+    num_missing_features = (4 - (param_zeros.shape[1] % 4)) % 4
+    if num_missing_features != 0:
+        param_zeros = np.pad(
+            param_zeros, ((0, 0), (0, num_missing_features)), constant_values=True
+        )
+
+    # Group into blocks and count full blocks
+    param_blocks = np.reshape(param_zeros, (-1, 4))
+    num_zeros_per_block = np.count_nonzero(param_blocks, axis=1)
+    num_zero_blocks = np.count_nonzero(num_zeros_per_block == 4, axis=0)
+
+    return num_zero_blocks, param_blocks.shape[0]
+
+
+def get_zero_point(model: ModelProto, node: NodeProto) -> int:
+    """
+    :param model: model in which node's zero point initializer exists
+    :param node: node to find zero point of
+    :return: zero point of given node
+    """
+    if is_quantized_layer(node):
+        initializer_name = node.input[3]
+        initializer = get_initializer(model, initializer_name)
+        zero_point_initializer = numpy_helper.to_array(initializer)
```

Comment:
nit: rename to `zero_point_array` or just `zero_point` since it's been converted to a numpy array